ytmusicapi/ytmusic.py

```diff
@@ -121,7 +129,12 @@ def __init__(self,
             except KeyError:
                 raise Exception("Your cookie is missing the required value __Secure-3PAPISID")
 
+
+
     def _send_request(self, endpoint: str, body: Dict, additionalParams: str = "") -> Dict:
+
+        if self.is_oauth_auth and self.ytmusic_oauth is not None:
+            self.headers = self.ytmusic_oauth.load_headers(dict(self.input_dict), self.auth)
```

Comment:
with my suggestion you can instead call `prepare_headers` here and remove the second condition in the `if`